pkg/checks/dns/dns_test.go

```diff
@@ -0,0 +1,350 @@
+			ctx:     context.Background(),
+			want: types.Result{
+				Data: map[string]Result{},
+			},
+		},
+		{
+			name: "success with one target lookup",
+			mockSetup: func() *DNS {
+				c := newCommonDNS()
+				c.client = &ResolverMock{
+					LookupHostFunc: func(ctx context.Context, addr string) ([]string, error) {
+						return []string{exampleIP}, nil
+					},
+					SetDialerFunc: func(d *net.Dialer) {},
+				}
+				return c
+			},
+			targets: []string{exampleURL},
+			ctx:     context.Background(),
+			want: types.Result{
+				Data: map[string]Result{
+					exampleURL: {Resolved: []string{exampleIP}},
+				},
+			},
+		},
+		{ //nolint:dupl // normal lookup
+			name: "success with multiple target lookups",
+			mockSetup: func() *DNS {
+				c := newCommonDNS()
+				c.client = &ResolverMock{
+					LookupHostFunc: func(ctx context.Context, addr string) ([]string, error) {
+						return []string{exampleIP, sparrowIP}, nil
+					},
+					SetDialerFunc: func(d *net.Dialer) {},
+				}
+				return c
+			},
+			targets: []string{exampleURL, sparrowURL},
+			ctx:     context.Background(),
+			want: types.Result{
+				Data: map[string]Result{
+					exampleURL: {Resolved: []string{exampleIP, sparrowIP}},
+					sparrowURL: {Resolved: []string{exampleIP, sparrowIP}},
+				},
+			},
+		},
+		{ //nolint:dupl // reverse lookup
+			name: "success with multiple target reverse lookups",
+			mockSetup: func() *DNS {
+				c := newCommonDNS()
+				c.client = &ResolverMock{
+					LookupAddrFunc: func(ctx context.Context, addr string) ([]string, error) {
+						return []string{exampleURL, sparrowURL}, nil
+					},
+					SetDialerFunc: func(d *net.Dialer) {},
+				}
+				return c
+			},
+			targets: []string{exampleIP, sparrowIP},
+			ctx:     context.Background(),
+			want: types.Result{
+				Data: map[string]Result{
+					exampleIP: {Resolved: []string{exampleURL, sparrowURL}},
+					sparrowIP: {Resolved: []string{exampleURL, sparrowURL}},
+				},
+			},
+		},
+		{
+			name: "error - lookup failure for a target",
+			mockSetup: func() *DNS {
+				c := newCommonDNS()
+				c.client = &ResolverMock{
+					LookupHostFunc: func(ctx context.Context, addr string) ([]string, error) {
+						return nil, fmt.Errorf("lookup failed")
+					},
+					SetDialerFunc: func(d *net.Dialer) {},
+				}
+				return c
+			},
+			targets: []string{exampleURL},
+			ctx:     context.Background(),
+			want: types.Result{
+				Data: map[string]Result{
+					exampleURL: {Error: stringPointer("lookup failed")},
+				},
+			},
+		},
+		{
+			name: "error - timeout scenario for a target",
+			mockSetup: func() *DNS {
+				c := newCommonDNS()
+				c.client = &ResolverMock{
+					LookupHostFunc: func(ctx context.Context, addr string) ([]string, error) {
+						time.Sleep(6 * time.Second)
+						return nil, fmt.Errorf("context deadline exceeded")
+					},
+					SetDialerFunc: func(d *net.Dialer) {},
+				}
+				return c
+			},
+			targets: []string{exampleURL},
+			ctx:     context.Background(),
```

Comment:
I would create a new context in each test instead of adding it in the test cases, since it's always the same. Nothing to be gained from having the same parameter in each test case, when it doesn't change at all.